File: week3/atakantekatan/74. Search a 2D Matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        if target < matrix[0][0] or target > matrix[-1][-1]:
            return False
        l, r = 0 , len(matrix[0]) - 1
        row = 0
        while target > matrix[row][r]:
            row += 1
        while l <= r:
            mid = (l+r) // 2
            
            if target < matrix[row][mid]:
                r = mid - 1
            elif target > matrix[row][mid]:
                l = mid + 1
            elif target == matrix[row][mid]:
                return True 
        return False

File: week3/Can-Yavuz/74-search-a-2D-matrix.cs
public class Solution {
    public bool SearchMatrix(int[][] matrix, int target) {

        if (matrix == null || matrix.Length == 0 || matrix[0].Length == 0) return false;

       int rows = matrix.Length;
       int cols = matrix[0].Length;
    
        int first = 0;
        int last = cols * rows -1;
        while(first <= last){
        int mid = first + (last - first) / 2;
        int midVal = matrix[mid / cols][mid % cols];
        if(midVal == target){
            return true;
        }else if(midVal  < target){
            first = mid +1;
        }else
            last = mid -1;
        }
        return false;
    }
}

